src/logic.js

```diff
@@ -501,7 +501,46 @@ define([
                 .map(function(ref) {
                     return [ref.path, null];
                 }).object().value();
-        }
+        },
+        /*
+         * Returns object describing references used in the logic manager
+         *
+         * {
+         *   /path/of/referenced/mug: {
+         *     /path/of/mug/referencing: property where ref occurs
+         *   }
+         * }
+         *
+         * Optionally can filter based on ufid or hashtag path
+         */
+        findUsages: function (ufidOrPath) {
+            var _this = this,
+                form = _this.form,
+                tableData = {},
+                refs = _this.reverse;
+
+            if (ufidOrPath && refs[ufidOrPath]) {
+                refs = refs[ufidOrPath];
+            }
+
+            _.each(refs, function (value, key) {
+                var usedMug = form.getMugByUFID(key),
+                    mugReferences = {};
+                if (ufidOrPath && ufidOrPath === usedMug.hashtagPath) {
+                    return;
+                }
+                _.each(value, function (value, key) {
+                    _.each(value, function (value) {
+                        var usedInMug = form.getMugByUFID(key),
+                            readablePropName = usedInMug.spec[value.property].lstring;
```

Comment:
Arguably should return `value.property` and the modal code should change it to a readable name, but I think this is good enough